Initial changes for updated appointment list

vbahinwillit opened the pull request:
## Description
This PR updates the appointment list display.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#35271


## Testing done
Unit and e2e testing

## Screenshots
**Upcoming**
![localhost_3001_health-care_schedule-view-va-appointments_appointments_ (3)](https://user-images.githubusercontent.com/3587066/152645378-b59119f0-2017-47c0-bd60-29f89ae8e42b.png)

---
**Pending**
![localhost_3001_health-care_schedule-view-va-appointments_appointments_ (4)](https://user-images.githubusercontent.com/3587066/152645414-a5b14b92-3911-4913-b836-cd732ba574a9.png)

---
**Past**
![localhost_3001_health-care_schedule-view-va-appointments_appointments_ (6)](https://user-images.githubusercontent.com/3587066/152645516-b0354d2d-3b1d-4b0d-8bb5-8edbfd3ef226.png)


## Acceptance criteria
- [ ] All test should pass

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

vbahinwillit committed:
Initial changes for updated appointment list

jenniemc commented on src/applications/vaos/appointment-list/components/AppointmentsPageV2/index.jsx:
```diff
@@ -62,17 +67,25 @@ export default function AppointmentsPageV2() {
       document.title = `${subPageTitle} | ${pageTitle} | Veterans Affairs`;
     },
-    [subPageTitle],
+    [subPageTitle, featureStatusImprovement],
```
I think we can exclude `featureStatusImprovement` from re-rending the page since this is a toggle flag boolean that remains static.
```suggestion
    [subPageTitle],
```

vbahinwillit replied:
ESLint was complaining when missing.

jenniemc reviewed (changes requested):
The initial changes look really good. 
Noticed that all the screenshots have the identical breadcrumb but from the design specs the [Pending prototype](https://www.sketch.com/s/7ceea77b-e368-4ed1-af98-3b7c57cf3124/a/WKQoDDw) and [Past appt prototype](https://www.sketch.com/s/7ceea77b-e368-4ed1-af98-3b7c57cf3124/a/52J1OOL) should have the last linkable breadcrumb going to "Your Appointment"

jenniemc commented:
I played around with PageLayout.jsx and AppointmentsPageV2 and was able to altered the breadcrumb based on which appointment status was selected. 

On the  AppointmentsPageV2, I passed in the `featureStatusImprovement` and the `subHeading` values to the the PageLayout component
```    
   <PageLayout
      showBreadcrumbs
      showNeedHelp
      featureStatusImprovement={featureStatusImprovement}
      subHeading={subHeading}
    >
```
Then on the PageLayout component, I look the boolean value to write out conditions

```
export default function PageLayout({
  children,
  showBreadcrumbs,
  showNeedHelp,
  featureStatusImprovement,
  subHeading,
}) {
  return (
    <div className="vads-l-grid-container vads-u-padding-x--2p5 large-screen:vads-u-padding-x--0 vads-u-padding-bottom--2p5">
      {(showBreadcrumbs && !featureStatusImprovement) ||
      (showBreadcrumbs &&
        featureStatusImprovement &&
        subHeading === 'Your appointments') ? (
        <Breadcrumbs />
      ) : (
        <Breadcrumbs>
          <Link to="/">{subHeading}</Link>
        </Breadcrumbs>
      )}
      <div className="vads-l-row">
        <div className="vads-l-col--12 medium-screen:vads-l-col--8 vads-u-margin-bottom--2">
          {children}
          {showNeedHelp && <NeedHelp />}
        </div>
      </div>
    </div>
  );
}

```

vbahinwillit committed:
Update breadcrumbs when clicking links

vbahinwillit committed:
Update title to 'Pending'

vbahinwillit commented:
> 

Yea, talked to Peter about that and it was decided to update the breadcrumbs accordingly.

jenniemc reviewed (changes requested):
Looks like the breadcrumb text is changing to reflect status selected. The downside is when I drill down to an appointment request detail page then click on the "pending" link on the breadcrumb, I get an error message. 

![image](https://user-images.githubusercontent.com/54327023/152177749-82c433f8-98ff-4518-ae84-ae8ec31c0624.png)
The pending link is going to a route that doesn't exists.  `http://localhost:3001/health-care/schedule-view-va-appointments/appointments/requests/requested`

jenniemc commented on src/applications/vaos/appointment-list/redux/actions.js:
```diff
@@ -731,3 +732,9 @@ export function fetchFacilitySettings() {
 }
+
+export function updateBreadcrumb(breadcrumb) {
+  return async (dispatch, _getState) => {
```
Just curious why use `_getState`? Other asynchronous calls have `getState` as the second param.

vbahinwillit replied:
'getState' is not being used so I could have deleted it. If you add an '_' before unused parameters eslint will ignore them. Sometimes I leave the unused parameters as a reminder.

vbahinwillit committed:
Fixed broken breadcrumb links

vbahinwillit committed:
Merge branch 'master' into 35271-vaos-status-homepage-update

vbahinwillit committed:
Fixed failing e2e test
Added seperate handler functions to deal with eslint react/jsx-no-bind
Fixed all auto fixable eslint warnings

vbahinwillit committed:
Fixed errors

vbahinwillit committed:
Modified per code review

vbahinwillit committed:
Add unit test

jenniemc commented on src/applications/vaos/appointment-list/components/AppointmentListNavigation.jsx:
```diff
@@ -0,0 +1,86 @@
+import { selectFeatureStatusImprovement } from '../../redux/selectors';
+
+function handleClick({ id, history, dispatch, callback }) {
+  return _event => {
```
Just curious why use `_event`?

vbahinwillit replied:
I'll delete it since it's not used.

jenniemc commented on src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentListItem.jsx:
```diff
@@ -67,14 +110,13 @@ export default function AppointmentListItem({ appointment, facility }) {
     >
+      {/* Disabling for now since add role=button and tab=0 fails another accessiblity check: */}
+      {/* Nested interactive controls are not announced by screen readers */}
+      {/* eslint-disable-next-line jsx-a11y/no-static-element-interactions */}
```
If not this PR, we will have to bring back  `tabIndex="0"` somehow to address accessibility. Removing it will cause the UI to scroll to the bottom of the screen if veteran uses the spacebar to activate display appointment details page. 

One positive note is that using the `Return` key works as expected.

jenniemc commented on src/applications/vaos/appointment-list/components/AppointmentsPageV2/index.jsx:
```diff
@@ -37,49 +41,88 @@ function getDropdownValueFromLocation(pathname) {
+        let prefix = subPageTitle;
+        if (subPageTitle.startsWith('Request')) prefix = 'Pending appointments';
+        document.title = `${prefix} | VA online scheduling | Veterans Affairs`;
+        pageTitle = 'Your appointments';
```
Helps with usability to bring focus to the h1 heading rather than document.body
```suggestion
        pageTitle = 'Your appointments';
        scrollAndFocus('h1');
```

vbahinwillit replied:
Good catch!

jenniemc commented on src/applications/vaos/appointment-list/components/AppointmentsPageV2/index.jsx:
```diff
@@ -37,49 +41,88 @@ function getDropdownValueFromLocation(pathname) {
+        document.title = `${prefix} | VA online scheduling | Veterans Affairs`;
+        pageTitle = 'Your appointments';
+      } else {
+        document.title = `${subPageTitle} | ${pageTitle} | Veterans Affairs`;
```
Same as above
```suggestion
        document.title = `${subPageTitle} | ${pageTitle} | Veterans Affairs`;
        scrollAndFocus('h1');
```

jenniemc reviewed (approved):
LGTM! 

Just have a few minor suggestions.  Mainly to address the accessibility.

jenniemc commented:
Currently we've only adjusted print mode for upcoming appointment list. If veteran wants to print pending and past appointment lists, this is how it looks. 

Maybe we can print the `<h1>` on print media to distinguish each of the different status. I suppose that is a discussion with @outerpress .

Pending Appt list
![image](https://user-images.githubusercontent.com/54327023/152866223-382d8b4c-1371-4c0c-ba7f-71602a58831b.png)

Past Appt list
![image](https://user-images.githubusercontent.com/54327023/152866170-140040dd-c035-4c91-9a6e-2ae8cdc98904.png)

vbahinwillit committed:
Modified per code review

* Removed unsed event variable
* Added scrollAndFocus for accessibility

vbahinwillit committed:
Merge branch 'master' into 35271-vaos-status-homepage-update
